### PR TITLE
Refactoring/aws/cloudwatch

### PR DESCRIPTION
**Pull request by Aboisier**
Migrated CloudWatch to the new architecture, see #183 .

### CODE REVIEW

File: ScoutSuite/providers/aws/facade/cloudwatch.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from ScoutSuite.providers.aws.facade.utils import AWSFacadeUtils
+
+
+class CloudWatch:
+    def get_alarms(self, region):
```
Comment (misg) [inline ScoutSuite/providers/aws/facade/cloudwatch.py]:
```suggestion
    async def get_alarms(self, region):
```

File: ScoutSuite/providers/aws/facade/cloudwatch.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+        return AWSFacadeUtils.get_all_pages('cloudwatch', region, 'describe_alarms', 'MetricAlarms')
```
Comment (misg) [inline ScoutSuite/providers/aws/facade/cloudwatch.py]:
```suggestion
        return await AWSFacadeUtils.get_all_pages('cloudwatch', region, 'describe_alarms', 'MetricAlarms')
```

File: ScoutSuite/providers/aws/resources/cloudwatch/service.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+from ScoutSuite.providers.aws.facade.facade import AWSFacade
+from ScoutSuite.providers.aws.resources.regions import Regions
+from ScoutSuite.providers.aws.resources.resources import AWSResources
+
+from ScoutSuite.providers.utils import get_non_provider_id
+
+
+class Alarms(AWSResources):
+    async def fetch_all(self, **kwargs):
+        raw_alarms = self.facade.cloudwatch.get_alarms(self.scope['region'])
```
Comment (misg) [inline ScoutSuite/providers/aws/resources/cloudwatch/service.py]:
```suggestion
        raw_alarms = await self.facade.cloudwatch.get_alarms(self.scope['region'])
```

File: ScoutSuite/providers/aws/resources/cloudwatch/service.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            name, resource = self._parse(raw_alarm)
+            self[name] = resource
+
+    def _parse(self, raw_alarm):
+        """
+        Parse a single CloudWatch trail
+
+        :param global_params:           Parameters shared for all regions
+        :param region:                  Name of the AWS region
+        :param alarm:                   Alarm
```
Comment (Remi05) [inline ScoutSuite/providers/aws/resources/cloudwatch/service.py]: The params description does not seem to be up to date.

**Pull request merged**